src/site/_includes/components/CodePattern.js

```diff
@@ -0,0 +1,66 @@
+const {html} = require('common-tags');
+const Prism = require('prismjs');
+
+module.exports = () => {
+  const pattern = {
+    id: 'example-pattern',
+    layout: 'pattern',
+    title: 'An example pattern in a suite',
+    description: 'A description for the example pattern',
+    content: '<h2>An example pattern in a suites</h2>',
+    assets: {
+      'body.html': {
+        content: '<h1>Hello</h1>',
+        type: 'html',
+      },
+      'script.js': {
+        content: 'document.write("hello")',
+        type: 'js',
+      },
+      'style.css': {
+        content: 'background: red;',
+        type: 'css',
+      },
+    },
+    demo: '/patterns/example-pattern/demo.html',
+    suite: '/patterns/',
+  };
+
+  const assets = Object.values(pattern.assets)
+    .map((asset) => {
+      const content = Prism.highlight(
+        asset.content,
+        Prism.languages.html,
```

Comment:
should this be Prism.languages[asset.type] ? Possibly done in some kind of try/catch with a fallback to 'text' if there's an error?